components/server/src/ide-config.ts

```diff
@@ -190,6 +191,20 @@ export class IDEConfigService {
                 }
             }
 
+            for (const [id, option] of Object.entries(newValue.ideOptions.options).filter(([_, x]) => x.latestImage)) {
+                try {
+                    value.ideOptions.options[id].latestImage = await this.resolveImageDigest(option.latestImage!);
```

Comment:
Nit: I would find it a bit "cleaner" to use a boolean like `option.useLatestImage`, and if it's true, set `option.latestImage` (just like `option.resolveImageDigest` and `option.image` above).

Here, we seem to use `option.latestImage` both as a "boolean" (i.e. `filter(([_, x]) => x.latestImage))` above) _and_ a resolved value `.latestImage = await ...`. This looks a little bit dangerous (e.g. we might accidentally resolve the same image multiple times, or maybe forget to resolve it elsewhere, etc).

However, not blocking the review due to this. Please feel free to address in a follow-up PR.

Reply:
Thanks @iQQBot! Yes, I understand this, but my point was, I prefer something like this:

```ts
type Option = {
  resolveLatestImage: boolean;
  latestImage?: string;
}

if (option.resolveLatestImage) {
  option.latestImage = await resolve();
}
```

where the controlling condition is different from what's actually resolved.

Currently, the code seems to work like this:

```ts
type Option = {
  latestImage?: string;
}

if (option.latestImage) {
  option.latestImage = await resolve();
}
```

This makes it a bit unclear if an image has already been resolved or not, e.g.:
- If the code is called multiple times, you might accidentally resolve this image multiple times
- You cannot do `if (!option.latestImage) { option.latestImage = await resolve(); }`
- Someone might accidentally "forget" to resolve the image in some cases (i.e. leave it as `option.latestImage === "latest"`, because `if (option.latestImage)` looks "true")